corehq/apps/app_manager/models.py

```diff
@@ -6301,13 +6314,7 @@ def _check_modules(self):
         if not self.modules:
             errors.append({'type': "no modules"})
         for module in self.get_modules():
-            try:
-                errors.extend(module.validate_for_build())
-            except ModuleNotFoundException as ex:
-                errors.append({
-                    "type": "missing module",
-                    "message": six.text_type(ex)
-                })
```

Comment:
Can you walk me through how moving this out to `ModuleBase` makes it go from appearing only on the built do appearing on the module page? I feel like I'm missing something important

Reply:
Yep: this PR moves the missing module check from being an app-level error to a module-level error.

The errors displayed on the module page are those found by `ModuleBase.validate_for_build`: https://github.com/dimagi/commcare-hq/blob/2354705b90f782c7721837bc3a895e27624d85c9/corehq/apps/app_manager/views/modules.py#L939-L942  Prior to this PR, the try/catch for `ModuleNotFoundException` and therefore the "missing module" error was only part of `_check_modules`, which is only called by `validate_app` so it only showed when making a build on the releases page.

Reply:
Hi Danny, did this explain well enough? I'd like to get this ticket closed out.

Reply:
Oh I see, I had it backwards. I thought you were moving this "out" (i.e. from callee function to caller function), but I see now after your explanation that you were moving it "in". Makes total sense. Thanks for the help